src/components/DataViewer/index.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  createStyles,
+  Theme,
+  Grid,
+  Paper,
+  WithStyles,
+  withStyles,
+  IconButton,
+  Button,
+  CircularProgress,
+} from '@material-ui/core';
+import { Close } from '@material-ui/icons';
+import {
+  datasetSelector,
+  loadingDatasetSelector,
+  clearDataset,
+  loadDataset,
+  DatasetParams,
+  updateAdminId,
+} from '../../context/datasetStateSlice';
+import { dateRangeSelector } from '../../context/mapStateSlice/selectors';
+import Chart from '../DataDrawer/Chart';
+import { ChartConfig } from '../../config/types';
+
+function DataViewer({ classes }: DatasetProps) {
+  const dispatch = useDispatch();
+  const isDatasetLoading = useSelector(loadingDatasetSelector);
+  const { startDate: selectedDate } = useSelector(dateRangeSelector);
+
+  const {
+    data: dataset,
+    boundaryProps,
+    id,
+    serverParams,
+    title,
+    chartType,
+  } = useSelector(datasetSelector);
+
+  useEffect(() => {
+    if (boundaryProps && serverParams && id && selectedDate) {
+      const datasetParams: DatasetParams = {
+        id,
+        boundaryProps,
+        url: serverParams.url,
+        serverLayerName: serverParams.layerName,
+        selectedDate,
+      };
+
+      dispatch(loadDataset(datasetParams));
+    }
+  }, [id, boundaryProps, dispatch, serverParams, selectedDate]);
+
+  if (
```

Comment:
It seems to odd to have to check all of these. If dataset is not null, surely the other datasetSelector items are not null?

Maybe you could simplify the logic. Otherwise let's use a list and "some" or "all" here to make it more readable